src/include/kdberrors.h

```diff
@@ -0,0 +1,313 @@
+/**
+ * @file
+ *
+ * @brief
+ *
+ * @copyright BSD License (see doc/LICENSE.md or http://www.libelektra.org)
+ */
+
+#ifndef KDBERRORS_H
+#define KDBERRORS_H
+
+#include <kdb.h>
+#include <kdbhelper.h>
+#include <kdblogger.h>
+#include <kdbmacros.h>
+
+#ifdef __cplusplus
+extern "C" {
+using Key = ckdb::Key;
```

Comment:
No, I have tried it but failed due to my insufficient lack of C/C++ skills.

Reply:
Removing the `ckdb` namespace should be pretty easy, but it should be a separate PR since it affects many files. I can do it after #3115, if no one else has the time/knowledge.

Reply:
Maybe @Piankero can try again, 90% of the effort should be to simply remove all occurrences of the text string "ckdb".

Reply:
yeah sure, I can write that regex remove those occurrences and open a PR for @kodebach to continue?

Reply:
Yes, it would be nice if you could start the PR. Just using a regex won't work since you cannot correctly remove the closing brackets corresponding to `namespace ckdb {` declarations. But that shouldn't be too much effort to do manually.

Reply:
no problem, I will create it in your repository tomorrow morning.
Just to be sure:
I should remove all `ckdb::` as well as all `namespace ckdb {` + `}` occurrences?

Reply:
> I should remove all ckdb:: as well as all namespace ckdb { + } occurrences?

The safest thing would be too replace `ckdb::` with just `::`, e.g. replacing `ckdb::Key` to `::Key`. That way we explicitly refer to the global namespace and shouldn't run into any problems, if someone used for example `using namespace kdb;` and then `ckdb::Key`.